libs/base/ui/overlays/notification-center/src/registry.controller.ts

```diff
@@ -1,142 +1,110 @@
-        if (!item.key) {
-          //get last key or generate new one
-          const key = createKey(this.registry);
-
-          const mergedStrategy = {
-            ...defaultStrategy,
-            ...item,
-          };
-
-          //add auto-close handler
-          if (mergedStrategy.autoClose) {
-            this._delayedCallbacks[key] = setTimeout(() => {
-              this.toggleVisibility(key);
-            }, mergedStrategy.autoCloseTime);
-          }
-          return {
-            ...mergedStrategy,
-            key,
-            visible: false,
-          };
-        }
-
-        return item;
-      }
-    );
+import { AlertType } from '@spryker-oryx/ui';
+import { Notification } from '../../notification';
 
-    this._registry = newRegistryWithParams;
-    this.host.requestUpdate();
-  }
+const DESTROY_DELAY_TIME = 600;
 
-  get registry(): NotificationRegistry[] {
-    return this._registry;
+export class RegistryController implements ReactiveController {
+  protected items: NotificationRegistry[] = [];
+  protected autoCloseQueue: {
+    [key: string]: {
+      autoCloseTime?: number;
+      timeout: ReturnType<typeof setTimeout>;
+    };
+  } = {};
+
+  getItems(): NotificationRegistry[] {
+    return this.items;
   }
 
-  preventAutoClose(targetKey: string): void {
-    if (this.notificationIsVisible(targetKey)) {
-      clearTimeout(this._delayedCallbacks[targetKey]);
-      return;
+  add(item: Notification): void {
+    const strategy = this.resolveStrategy(item);
+    const registryItem: NotificationRegistry = {
+      key: Date.now(),
+      ...strategy,
+    };
+    if (registryItem.autoClose) {
+      this.scheduleAutoClosing(registryItem);
     }
-
-    this.toggleVisibility(targetKey);
+    this.items.unshift(registryItem);
+    this.host.requestUpdate();
+    setTimeout(() => {
+      registryItem.visible = true;
+      this.host.requestUpdate();
+    }, 0);
   }
 
-  handleNotificationClose(key: string): void {
-    this.removeFromRegistry(key, true);
+  protected scheduleAutoClosing(item: NotificationRegistry): void {
+    // const key = item.key as string;
+    this.autoCloseQueue[item.key] = {
+      autoCloseTime: item.autoCloseTime,
+      timeout: setTimeout(() => {
+        this.destroy(item.key);
+      }, item.autoCloseTime),
+    };
   }
 
-  handleTransitionEnd(key: string): void {
-    if (!this.notificationIsVisible(key)) {
-      this.removeFromRegistry(key);
+  protected resolveStrategy(item: Notification): Notification {
+    const strategy = { floating: true, closable: true, ...item };
+
+    if (strategy.autoClose === undefined) {
+      strategy.autoClose =
+        strategy.type === AlertType.Info || AlertType.Success === strategy.type;
     }
-  }
 
-  protected removeFromRegistry(targetKey: string, force = false): void {
-    if (force) {
-      //should clear the delayed callback
-      this.preventAutoClose(targetKey);
+    if (!strategy.closable && !strategy.autoClose) {
```

Comment:
Why do we need this resolving? If I'm not mistaken our notifications are supporting custom content where can be custom close button. But this logic prevents having non autoclosable notification without general close button.

I believe that will be enough just this:
`const strategy = { floating: true, closable: true, autoClose: item.type === AlertType.Info || item.type === AlertType.Success, ...item };`